p2p/protocol/identify/obsaddr.go

```diff
@@ -200,6 +207,21 @@ func (oas *ObservedAddrManager) filter(observedAddrs []*observedAddr) []ma.Multi
 		}
 	}
 
+	// For certain use cases such as hole punching, it's better to advertise even unactivated observed addresses rather than none at all
```

Comment:
@vyzo This needs your attention. I discussed this with @Stebalien and he was okay with it.